lioncore/metametamodel/lioncore.json

```diff
@@ -0,0 +1,2180 @@
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "NamespacedEntity-key"
+          },
+          "value": "Language-version"
+        }
+      ],
+      "children": [],
+      "references": [
+        {
+          "reference": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Property-type"
+          },
+          "targets": [
+            {
+              "resolveInfo": "String",
+              "reference": "LIonCore-M3-String"
+            }
+          ]
+        }
+      ],
+      "parent": "-id-Language"
+    },
+    {
+      "id": "-id-LanguageElement",
+      "concept": {
+        "language": "LIonCore-M3",
+        "version": "1",
+        "key": "Concept"
+      },
+      "properties": [
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Concept-abstract"
+          },
+          "value": "true"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Concept-partition"
+          },
+          "value": "false"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "LIonCore-builtins-INamed-name"
+          },
+          "value": "LanguageElement"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "NamespacedEntity-key"
+          },
+          "value": "LanguageElement"
+        }
+      ],
+      "children": [
+        {
+          "containment": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "FeaturesContainer-features"
+          },
+          "children": []
+        }
+      ],
+      "references": [
+        {
+          "reference": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Concept-extends"
+          },
+          "targets": []
+        },
+        {
+          "reference": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Concept-implements"
+          },
+          "targets": [
+            {
+              "resolveInfo": "NamespacedEntity",
+              "reference": "-id-NamespacedEntity"
+            }
+          ]
+        }
+      ],
+      "parent": "-id-LIonCore-M3"
+    },
+    {
+      "id": "-id-NamespacedEntity",
+      "concept": {
+        "language": "LIonCore-M3",
+        "version": "1",
+        "key": "ConceptInterface"
+      },
+      "properties": [
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "LIonCore-builtins-INamed-name"
+          },
+          "value": "NamespacedEntity"
+        },
+        {
+          "property": {
```

Comment:
Is Key a property of NamespacedEntity? I was not expecting this

In the Java implementation I have an interface called HasKey, but we discussed the fact that it should be just an implementation detail, so we do not have a common point where to define the key.

I think that NamespacedEntity could not play this role, as the Language has a key but it is not a NamespacedEntity.

I think that we have two options:
- either we use two key properties (one declared under Language and one under LanguageElement)
- or we introduce a common ancestore for Language and LanguageElement which defines the key

Reply:
Currently option 1 is implemented here. Option 2 would be `HasKey`. As you say, we decided against `HasKey`, which would leave Option 1.
Conceptually, I agree `key` is a bit out of place within `NamespacedEntity` -- `key` is a language thing, namespaces could be used outside this context.
We could rediscuss this tomorrow?

Reply:
Sure, makes sense